assignments/lambda-classes.js

```diff
@@ -1 +1,212 @@
 // CODE here for your Lambda Classes
+//Person Constructor
+
+class Person {
+    constructor(props){
+    this.name = props.name;
+    this.age = props.age;
+    this.location = props.location;
+    this.gender = props.gender;
+}
+
+speak(){
+    return `Hello my name is ${this.name}, I am from ${this.location}.`;
+}
+}
+
+//Instructor Constructor
+class Instructor extends Person {
+    constructor(props){
+        super(props);
+        this.specialty = props.specialty;
+        this.favLanguage = props.favLanguage;
+        this.catchPhrase = props.catchPhrase;
+    
+    }
+
+    gradeBlaster(student){
+        let addOrSub = Math.random()* 3;
+        let randomPoints = Math.floor(Math.random()* 20);
+        let newGrade; 
+        if(addOrSub > 1){
+            newGrade = student.grade += randomPoints;
+            if(newGrade > 100){
+                newGrade = 100;
+                
+            }
+            console.log(`${student.name}'s grade jumped to a ${newGrade}!`); 
+        }
+        else if(addOrSub < 1){
+            newGrade = student.grade -= randomPoints;
+            console.log(`${student.name}'s grade dropped to a ${newGrade}.`); 
+        }
+
+    }
+
+    demo(subject){
+        return `Today we are learning about ${subject}.`;
+    }
+
+    grade(student, subject){
+        return `${student.name} receives a perfect score on ${subject}.`;
+    }
+} 
+
+//Student Constructor
+class Student extends Person {constructor(props){
+    super(props);
+    this.previousBackground = props.previousBackground;
+    this.className = props.className;
+    this.favSubjects = props.favSubjects;
+    /* Below I meant set up a conditional that would prevent the grade prop from 
```

Comment:
It is not working with gradeBlaster() because You are not using constructor that time but instead rewriting the property to whatever gradeBlaster() returns. So You can either directly set the conditional on gradeBlaster() or try to play around with setters and getters for an object to configure what the grade can be.